src/DynamoCore/Core/PathManager.cs

```diff
@@ -0,0 +1,280 @@
+        /// 
+        /// </summary>
+        /// <param name="library">The initial library file path.</param>
+        /// <returns>Returns true if the requested file can be located, or false
+        /// otherwise.</returns>
+        /// 
+        public bool ResolveLibraryPath(ref string library)
+        {
+            if (File.Exists(library)) // Absolute path, we're done here.
+                return true;
+
+            library = LibrarySearchPaths(library).FirstOrDefault(File.Exists);
+            return library != default(string);
+        }
+
+        #endregion
+
+        #region Public Class Operational Methods
+
+        /// <summary>
+        /// Constructs an instance of PathManager object.
+        /// </summary>
+        /// <param name="pathResolver">Reference of an IPathResolver object that
+        /// supplies additional path information. This argument is optional.</param>
+        /// 
+        internal PathManager(IPathResolver pathResolver)
+        {
+            // This method is invoked in DynamoCore.dll, dynamoCorePath 
+            // represents the directory that contains DynamoCore.dll.
+            var dynamoCorePath = Assembly.GetExecutingAssembly().Location;
+            dynamoCoreDir = Path.GetDirectoryName(dynamoCorePath);
+
+            if (!File.Exists(Path.Combine(dynamoCoreDir, "DynamoCore.dll")))
+            {
+                throw new Exception("Dynamo's core path could not be found. " +
+                    "If you are running Dynamo from a test, try specifying the " +
+                    "Dynamo core location in the DynamoBasePath variable in " +
+                    "TestServices.dll.config.");
+            }
+
+            // Current user specific directories.
+            userDataDir = CreateFolder(GetUserDataFolder());
+
+            userDefinitions = CreateFolder(Path.Combine(userDataDir, "definitions"));
+            logDirectory = CreateFolder(Path.Combine(userDataDir, "Logs"));
+            packagesDirectory = CreateFolder(Path.Combine(userDataDir, "packages"));
+            preferenceFilePath = Path.Combine(userDataDir, "DynamoSettings.xml");
+
+            // Common directories.
+            commonDataDir = CreateFolder(GetCommonDataFolder());
+
+            commonDefinitions = CreateFolder(Path.Combine(commonDataDir, "definitions"));
+            samplesDirectory = GetSamplesFolder(commonDataDir);
+
+            nodeDirectories = new HashSet<string>
+            {
+                Path.Combine(dynamoCoreDir, "nodes")
+            };
+
+            preloadedLibraries = new HashSet<string>
+            {
+                "VMDataBridge.dll",
+                "ProtoGeometry.dll",
+                "DSCoreNodes.dll",
+                "DSOffice.dll",
+                "DSIronPython.dll",
+                "FunctionObject.ds",
+                "Optimize.ds",
+                "DynamoUnits.dll",
+                "Tessellation.dll",
+                "Analysis.dll"
+            };
```

Comment:
Yes @pboyer, but that will not be done as part of the immediate PR (whose aim is to maintain the largest stability around test results). I have made [another task that must be done this sprint](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6600) for this work, as I expect there will be more test failures when that gets in, and I'll go through the tests to fix them all.